src/DGtal/images/ArrayImageAdapter.h

```diff
@@ -0,0 +1,544 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file ArrayImageAdapter.h
+ * @author Roland Denis (\c roland.denis@univ-smb.fr )
+ * LAboratory of MAthematics - LAMA (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2015/06/19
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(ArrayImageAdapter_RECURSES)
+#error Recursive header files inclusion detected in ArrayImageAdapter.h
+#else // defined(ArrayImageAdapter_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define ArrayImageAdapter_RECURSES
+
+#if !defined ArrayImageAdapter_h
+/** Prevents repeated inclusion of headers. */
+#define ArrayImageAdapter_h
+
+#if __cplusplus < 201103L
+  #error ArrayImageAdapter.h requires C++11.
+#endif
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <boost/assert.hpp>
+#include <boost/static_assert.hpp>
+#include <boost/concept/assert.hpp>
+#include <boost/iterator/iterator_concepts.hpp>
+#include <iterator>
+#include <type_traits>
+
+#include <DGtal/base/Common.h>
+#include <DGtal/images/CConstImage.h>
+#include <DGtal/images/ArrayImageIterator.h>
+#include <DGtal/base/IteratorCompletion.h>
+#include <DGtal/kernel/domains/Linearizer.h>
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  /**
+   * @brief Aim: Image adapter for generic arrays with sub-domain view capability.
+   *
+   * Description of template class 'ArrayImageAdapter' <p>
+   * It is an empty class that is specialized for HyperRectDomain (see corresponding documentation page).
+   *
```

Comment:
I agree but I don't understand how to link to a templated class: Doxygen do not recognize it if I write `DGtal::ArrayImageAdapter< TArrayIterator, HyperRectDomain< TSpace > >`... 

The only way I see is to tell Doxygen to ignore the generic definition so that he only sees the specialized definition.